### PR TITLE
Update `round-based` dep

### PR DESCRIPTION
**Pull request by survived**
New round-based release has new exciting features. This PR updates dependency to the latest version and takes advantage of new features which allows us:

1. Make `cggmp21-keygen` fully `no_std` compatible
2. Provide sync API to carry out the protocols

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,3 +9,7 @@ members = [
 exclude = [
     "wasm/no_std",
 ]
+
+[patch.crates-io.round-based]
+git = "https://github.com/dfns/round-based"
+branch = "state-machine"
```
Comment (survived) [inline Cargo.toml]: to be removed

File: wasm/no_std/Cargo.toml [hunk 2]
```diff
@@ -3,6 +3,17 @@ name = "crates-compile-in-nostd-wasm"
...
+features = ["serde", "hd-wallets", "spof", "udigest"]
+
+[dependencies.cggmp21-keygen]
+path = "../../cggmp21-keygen"
+default-features = false
+features = ["hd-wallets", "state-machine"]
+
+[patch.crates-io.round-based]
+git = "https://github.com/dfns/round-based"
+branch = "state-machine"
```
Comment (survived) [inline wasm/no_std/Cargo.toml]: to be removed

File: Cargo.toml (hunk 1 above)
Comment (survived) [reply Cargo.toml]: changed to `m`

File: wasm/no_std/Cargo.toml (hunk 2 above)
Comment (survived) [reply wasm/no_std/Cargo.toml]: changed to `m`

**Comment by survived** [comment]
@maurges this uses the latest version of `round-based` from `m`

File: cggmp21-keygen/src/progress.rs
Comment (maurges) [inline cggmp21-keygen/src/progress.rs:1]: I hope one day github makes a better diff viewer

```
morj@mac:~/Projects/cggmp21> git difft m -- cggmp21-keygen/src/progress.rs | wc -l
      38
morj@mac:~/Projects/cggmp21> git diff m -- cggmp21-keygen/src/progress.rs | wc -l
     734
```

(difft for me is https://difftastic.wilfred.me.uk/)

File: package.json
Comment (maurges) [inline package.json:1]: What are JS package files for?

File: cggmp21-keygen/src/lib.rs [hunk 3]
```diff
@@ -219,6 +232,23 @@ where
         )
         .await
     }
+
+    /// Returns a state machine that can be used to carry out the key generation protocol
+    ///
+    /// See [`round_based::state_machine`] for details on how that can be done.
+    #[cfg(feature = "state-machine")]
+    pub fn into_state_machine<R>(
```
Comment (maurges) [inline cggmp21-keygen/src/lib.rs:240]: Why are most sync methods named like this, while signing and and presigs are named `${name}_sync`?

File: package.json
Comment (survived) [reply package.json:1]: I have no idea...

Comment (survived) [reply package.json:1]: deleted

File: cggmp21-keygen/src/progress.rs
Comment (survived) [reply cggmp21-keygen/src/progress.rs:1]: Nice! I gotta try it

File: cggmp21-keygen/src/lib.rs (hunk 3 above)
Comment (survived) [reply cggmp21-keygen/src/lib.rs:240]: because `start_sync` would be misleading, and `sign_into_state_machine` is just strange

Comment (maurges) [reply cggmp21-keygen/src/lib.rs:240]: I would go for `start_protocol_sync` or `sync_start`. But I don't mind the current naming, just had to confirm the intentions

**Comment by maurges** [comment]
Hm, check-readme is failing now

File: package.json
Comment (survived) [reply package.json:1]: Ah it was added because I use `markdown-toc` tool, which is distributed via npm

Comment (survived) [reply package.json:1]: Reverted deletion. Basically, package.json and the other file lock the version of the tool we use

**Comment by maurges** [comment]
Looks good. Remind me, do we wait until publishing round-based first for this repo?

**Comment by survived** [comment]
I wanted to delay round-based release, but now I'm thinking that it should be fine, so I'll release it now

**Comment by survived** [comment]
@maurges done

**Review by maurges — APPROVED** [review]

**Pull request merged**